context/custom_store.tsx

```diff
@@ -20,6 +20,7 @@ import {
 interface IState {
   user: IUser | null,
   isAuthenticated: boolean,
+  moderation_state: 'blocked' | 'approved' | 'pending' | null;
```

Comment:
there is no pending state, its called "pending_moderation"

```suggestion
  moderation_state: 'blocked' | 'approved' | 'pending_moderation' | null;
```